lib/csdl2openapi.js

```diff
@@ -1,2905 +1,2908 @@
+      return paths;
+    }
+
+    const propsOfType = model.propertiesOfStructuredType(elementType);
+
+    const ignore = propsOfType.filter(
+      ([, type]) =>
+        type.$Kind !== "NavigationProperty" &&
+        type.$Type &&
+        !type.$Type.startsWith("Edm.") &&
+        !model.element(type.$Type),
+    );
+
+    // Keep old logging
+    for (const entry of ignore) {
+      messages.push(`Unknown type for element: ${JSON.stringify(entry)}`);
+    }
+
+    const properties = propsOfType
+      .filter((entry) => entry[1].$Kind !== "NavigationProperty")
+      .filter((entry) => !ignore.includes(entry))
+      .map(entryToProperty({ path: prefix, typeRefChain: [] }));
+
+    for (let i = 0; i < properties.length; i++) {
+      const property = properties[i];
+      if (!property.isComplex) {
+        paths.push(property.path);
+        continue;
+      }
+
+      const typeRefChainTail =
+        property.typeRefChain[property.typeRefChain.length - 1];
+
+      // Allow full cycle to be shown (0) times
+      if (
+        property.typeRefChain.filter((_type) => _type === typeRefChainTail)
+          .length > 1
+      ) {
+        // messages.push(`Cycle detected ${property.typeRefChain.join("->")}`);
+        continue;
+      }
+
+      const expanded = Object.entries(property.properties)
+        .filter((property) => property[1].$Kind !== "NavigationProperty")
+        .map(entryToProperty(property));
+      properties.splice(i + 1, 0, ...expanded);
+    }
+
+    return paths;
+  }
+
+  function entryToProperty(parent) {
+    return function (entry) {
+      const key = entry[0];
+      const property = entry[1];
+      const propertyType = property.$Type && model.element(property.$Type);
+
+      if (
+        propertyType &&
+        propertyType.$Kind &&
+        propertyType.$Kind === "ComplexType"
+      ) {
+        return {
+          properties: model.propertiesMapOfStructuredType(propertyType),
+          path: `${parent.path}${key}/`,
+          typeRefChain: parent.typeRefChain.concat(property.$Type),
+          isComplex: true,
+        };
+      }
+
+      return {
+        properties: {},
+        path: `${parent.path}${key}`,
+        typeRefChain: [],
+        isComplex: false,
+      };
+    };
+  }
+
+  /**
+   * Add parameter for query option $search
+   * @param {Array} parameters Array of parameters to augment
+   * @param {string} target Target container child of path
+   * @param {object} restrictions Navigation property restrictions of navigation segment
+   */
+  function optionSearch(parameters, target, restrictions) {
+    const searchRestrictions =
+      restrictions.SearchRestrictions ||
+      (target && target[voc.Capabilities.SearchRestrictions]) ||
+      {};
+
+    if (
+      searchRestrictions.Searchable !== false ||
+      searchRestrictions.SearchSyntax
+    ) {
+      if (searchRestrictions[voc.Core.Description]) {
```

Comment:
```suggestion
      if (searchRestrictions[voc.Core.Description] || searchRestrictions.SearchSyntax) {
```
Produce "custom" parameter description also for custom search syntax and don't reference the reuse parameter which in turn links to the OData standard search syntax.